src/ddx_errors.f90

```diff
@@ -0,0 +1,62 @@
+module ddx_errors
+
+use ddx_definitions
+
+!> ddX type for containing error information
+type ddx_error_type
+    !> Flag for error codes
+    integer :: flag = 0
+    !> Error message log
+    integer :: max_length = 2047
+    character(len=2047) :: message
+    integer :: message_length = 1
+end type ddx_error_type
+
+contains
+
+subroutine print_error(error)
+    implicit none
+    type(ddx_error_type), intent(in) :: error
+    write(6, "(A)") error % message(1:error % message_length-2)
+end subroutine print_error
+
+subroutine check_error(error)
+    implicit none
+    type(ddx_error_type), intent(in) :: error
+    if (error % flag .ne. 0) then
+        call print_error(error)
+        stop error % flag
+    end if
+end subroutine check_error
+
+subroutine update_error(error, message)
+    implicit none
+    type(ddx_error_type), intent(inout) :: error
+    character(len=*) :: message
+    integer :: message_start, message_stop
+    ! update the error flag by summing one
+    error % flag = error % flag + 1
+    message_start = error % message_length
+    message_stop = message_start + len(message) + 3
+    ! update the message if there is still space left
+    if (message_stop .lt. error % max_length) then
```

Comment:
Would it not make more sense to copy all the characters you can. I imagine a case where you are adding two messages and the first is too long and the second fits. Then the order of messages can be scrambled, which is probably not what you want (might make debugging super hard).

Reply:
Good catch. I reworked the `update_error` routine.
I have also added a small test that checks if it works correctly: we don't want a segmentation fault when an error is being raised by something else.